homeassistant/components/guardian/__init__.py

```diff
@@ -213,20 +214,14 @@ class GuardianEntity(CoordinatorEntity):
     """Define a base Guardian entity."""
 
     def __init__(  # pylint: disable=super-init-not-called
-        self,
-        entry: ConfigEntry,
-        kind: str,
-        name: str,
-        device_class: str | None,
-        icon: str | None,
+        self, entry: ConfigEntry, description: EntityDescription
     ) -> None:
         """Initialize."""
-        self._attr_device_class = device_class
         self._attr_device_info = {"manufacturer": "Elexa"}
         self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: "Data provided by Elexa"}
-        self._attr_icon = icon
-        self._attr_name = name
         self._entry = entry
+        self._type = description.key
```

Comment:
Replace the use of `self._type` with `self.entity_description.key` in the code instead, removing the need for this variable

```suggestion
```